emitters/yaml_base_emitter.cpp

```diff
@@ -412,14 +402,19 @@ bool yaml_base_emitter::check_editable_scalar(const std::string& filepath,
         return true;
     }
 
-    if (expected_scalar != have_scalar) {
-        notify("value mismatch; have `" + have_scalar + "`, expected `" + expected_scalar + "`",
+    if (expected_scalar == tag_value_optional_k && have_scalar == tag_value_optional_k) {
+        result = have;
+        return false;
```

Comment:
Why return `false` here? If the result is what we have, shouldn't that be no error?